src/local/butler/reproduce.py

```diff
@@ -120,20 +137,56 @@ def _post(url, body, force_reauthorization=False):
 
 def _get_testcase(testcase_id):
   """Retrieve the json representation of the test case with the given id."""
-  response, content = _post(TESTCASE_URL, body={'testcaseId': testcase_id})
+  response, content = _http_request(
+      TESTCASE_INFO_URL, body={'testcaseId': testcase_id})
 
   # TODO(mbarbella): Handle this gracefully.
   if response.status != 200:
     raise Exception('Failed to get test case information.')
 
-  testcase_map = json.loads(content)
+  testcase_map = json_utils.loads(content)
   return SerializedTestcase(testcase_map)
 
 
-def _download_testcase(_):
+def _download_testcase(testcase_id, testcase):
   """Download the test case and return its path."""
-  # TODO(mbarbella): Implement this.
-  return '/tmp/blah'
+  response, content = _http_request(
+      TESTCASE_DOWNLOAD_URL.format(testcase_id=testcase_id), method='GET')
+  bot_absolute_filename = response['x-goog-meta-filename']
+
+  # Create a temporary directory where we can store the test case.
+  testcase_directory = os.path.join(
+      environment.get_value('ROOT_DIR'), 'current-testcase')
+  shell.create_directory(testcase_directory)
+  testcase_path = os.path.join(testcase_directory,
+                               os.path.basename(bot_absolute_filename))
+
+  utils.write_data_to_file(content, testcase_path)
+
+  # Unpack the test case if it's archived.
+  # TODO(mbarbella): Rewrite setup.unpack_testcase and share this code.
+  if testcase.minimized_keys and testcase.minimized_keys != 'NA':
+    mask = data_types.ArchiveStatus.MINIMIZED
+  else:
+    mask = data_types.ArchiveStatus.FUZZED
+
+  if testcase.archive_state & mask:
```

Comment:
This is actually a mask here. If both the original and minimized test cases are archives, archive_state will be FUZZED | MINIMIZED (we should probably rewrite this, but it would require migrating many test cases).

Reply:
ACK